.eslintrc.js

```diff
@@ -0,0 +1,52 @@
+module.exports = {
+  extends: [
+    require.resolve('./config/eslint'),
+    'plugin:shopify/typescript',
+    'plugin:shopify/react',
+    'plugin:shopify/jest',
+    'plugin:shopify/prettier',
+  ],
+  parserOptions: {
+    project: 'tsconfig.json',
+  },
+  overrides: [
+    {
+      files: ['*.ts', '*.tsx'],
+      rules: {
+        'import/no-extraneous-dependencies': 'error',
+        'import/extensions': 'off',
+        'jsx-a11y/control-has-associated-label': 'off',
+        'node/no-extraneous-require': 'off',
+        'import/no-cycle': 'off',
+        'jest/require-tothrow-message': 'off',
+        'callback-return': 'off',
+        'jest/no-if': 'off',
+        'import/named': 'off',
+        'func-style': 'off',
+        'react/display-name': 'off',
+        'shopify/restrict-full-import': ['error', 'lodash'],
+        'shopify/jsx-no-hardcoded-content': 'off',
+        'shopify/jest/no-vague-titles': [
+          'error',
+          {
+            allow: ['all'],
+          },
+        ],
+      },
+    },
+    {
+      files: [
+        '**/*/test/**/*.ts',
+        '**/*/test/**/*.tsx',
+        '**/*/tests/*.ts',
+        '**/*/tests/*.tsx',
+        '*.test.ts',
+        '*.test.tsx',
+      ],
+      rules: {
+        'shopify/jsx-no-hardcoded-content': 'off',
```

Comment:
Here we turn this rule off to prevent requiring packages such as `@shopify/react-testing` in ever react-based package.